becquerel/core/rebin.py

```diff
@@ -100,40 +101,32 @@ def _check_any_overlap(in_edges, out_edges):
             old:           └┴┴┴┴┘
             new:   └┴┴┴┘
     """
-    if np.any(in_edges[..., -1] <= out_edges[0]):
+    if (in_edges[..., -1] <= out_edges[0]).any():
         raise RebinError('Input edges are all smaller than output edges')
-    if np.any(in_edges[..., 0] >= out_edges[-1]):
+    if (in_edges[..., 0] >= out_edges[-1]).any():
         raise RebinError('Input edges are all larger than output edges')
 
 
-def _broadcast(arr, shape):
-    """
-    broadcast arr out to the first dimension in shape
-    specifically for the case 1D -> 2D
-    copy req'd: the readonly array doesn't work w/ numba
-    """
-    if (arr.ndim == 1) and (len(shape) == 2):
-        return np.copy(np.broadcast_to(arr, (shape[0], arr.shape[0])))
-    else:
-        return arr
-
-
-def _check_shape(arr0, arr1, arr0_name='array0', arr1_name='array1',
-                 edges=False):
-    arr0_shape = arr0.shape
-    arr1_shape = arr1.shape
-    if edges:
-        if len(arr1_shape) == 1:
-            arr1_shape = (arr1_shape[0] - 1,)
-        else:
-            arr1_shape = (arr1_shape[0], arr1_shape[1] - 1,)
-    if arr0_shape != arr1_shape:
-        raise RebinError(
-            '{}{} does not have a shape compatible with {}{}'.format(
-                arr0_name, arr0_shape, arr1_name, arr1_shape))
-
-
-@nb.jit(nb.f8(nb.f8, nb.f8, nb.f8, nb.f8), nopython=True)
+# no longer check shape because errors will pop up during xr.apply_ufunc.
+# too many cases to check for; explicit checks defeats the purpose of
+# using xr.apply_ufunc. Also, since xr.DataArrays use dimension names,
+# checking shapes do not make sense.
+# def _check_shape(arr0, arr1, arr0_name='array0', arr1_name='array1',
+#                  edges=False):
+#     arr0_shape = arr0.shape
+#     arr1_shape = arr1.shape
+#     if edges:
+#         if len(arr1_shape) == 1:
+#             arr1_shape = (arr1_shape[0] - 1,)
+#         else:
+#             arr1_shape = (arr1_shape[0], arr1_shape[1] - 1,)
+#     if arr0_shape != arr1_shape:
+#         raise RebinError(
+#             '{}{} does not have a shape compatible with {}{}'.format(
+#                 arr0_name, arr0_shape, arr1_name, arr1_shape))
```

Comment:
Maybe just delete entirely, instead of commenting out?

Reply:
Where's the 😂 emoji. On a more serious note, I'm cool with that too.